### PR TITLE
Update Alibi Explainers to 0.4.0 Release

### LINKED ISSUE #792: Update Alibi Explainer Library to 0.4.0
/kind bug

[Alibi](https://github.com/SeldonIO/alibi) has released 0.4.0 which has some breaking changes. Our current explainer wrapper has a setup.py which isn't fixed to 0.3.0:

https://github.com/kubeflow/kfserving/blob/3b282458eafe7fa423f7f4dcc34bd8f492c07884/python/alibiexplainer/setup.py#L36

We need to:

 * Fix Alibi explainer wrapper to 0.3.0 to keep existing functionality
 * Update and handle breaking changes for 0.4.0

### PR DESCRIPTION
**Pull request by cliveseldon**
Fixes #792 

 * Updates library for 0.4.0 release of [Alibi](https://github.com/SeldonIO/alibi/releases/tag/v0.4.0)
 * Adds tests for Anchors Tabular, Text and Images

### CODE REVIEW

**Comment by jinchihe** [comment]
@cliveseldon I noticed you already added that in ignore list, but still failed, I guess that's caused the syntax-error in `anchor_tabular.py` and `explainer.py`. Yeah, we need upgrade that. But for pylint issues in `test_anchor_tabular.py ` and `test_anchor_text.py `, that should be ignored, or disabled, or fixed. otherwise, guess that still be fail in Python 3.6+,
```
8d73/src/kubeflow/kfserving/python/alibiexplainer/tests/test_anchor_tabular.py 
cwd=/mnt/test-data-volume/kubeflow-kfserving-presubmit-e2e-803-493ba35-4881-8d73/src/kubeflow/kfserving/python
INFO|2020-05-06T18:04:44|util.py:61| Subprocess output:

INFO|2020-05-06T18:04:45|util.py:72| Using config file /mnt/test-data-volume/kubeflow-kfserving-presubmit-e2e-803-493ba35-4881-8d73/src/kubeflow/kfserving/python/.pylintrc
INFO|2020-05-06T18:04:45|util.py:72| ************* Module tests.test_anchor_tabular
INFO|2020-05-06T18:04:45|util.py:72| E:  1, 0: No name 'anchor_tabular' in module 'alibiexplainer' (no-name-in-module)
INFO|2020-05-06T18:04:45|util.py:72| E:  5, 0: No name 'model' in module 'sklearnserver' (no-name-in-module)
INFO|2020-05-06T18:04:45|util.py:72| I: 19, 8: Module 'kfserving' has not 'Storage' member, but source is unavailable. Consider adding this module to extension-pkg-whitelist if you want to perform analysis based on run-time introspection of living objects. (c-extension-no-member)
```


```
8d73/src/kubeflow/kfserving/python/alibiexplainer/tests/test_anchor_text.py 
cwd=/mnt/test-data-volume/kubeflow-kfserving-presubmit-e2e-803-493ba35-4881-8d73/src/kubeflow/kfserving/python
INFO|2020-05-06T18:04:38|util.py:61| Subprocess output:

INFO|2020-05-06T18:04:39|util.py:72| Using config file /mnt/test-data-volume/kubeflow-kfserving-presubmit-e2e-803-493ba35-4881-8d73/src/kubeflow/kfserving/python/.pylintrc
INFO|2020-05-06T18:04:39|util.py:72| ************* Module tests.test_anchor_text
INFO|2020-05-06T18:04:39|util.py:72| E:  4, 0: No name 'model' in module 'sklearnserver' (no-name-in-module)
```

**Comment by jinchihe** [comment]
@cliveseldon The test-worker has been updated/merged, you can rebase and retry now. Thanks.

File: docs/samples/explanation/alibi/imagenet/README.md [hunk 1]
```diff
@@ -23,15 +23,17 @@ spec:
     explainer:
       alibi:
         type: AnchorImages
-        storageUri: "gs://seldon-models/tfserving/imagenet/explainer"
+        storageUri: "gs://seldon-models/tfserving/imagenet/alibi/0.4.0"
         config:
           batch_size: "25"
+	  stop_in_first: "True"
```
Comment (yuzisun) [inline docs/samples/explanation/alibi/imagenet/README.md:29]: Any doc links on what this means?

File: python/alibiexplainer/alibiexplainer/explainer.py [hunk 2]
```diff
@@ -70,9 +74,14 @@ def _predict_fn(self, arr: Union[np.ndarray, List]) -> np.ndarray:
         return np.array(resp["predictions"])
 
     def explain(self, request: Dict) -> Any:
-        if self.method is ExplainerMethod.anchor_tabular or self.method is ExplainerMethod.anchor_images or self.method is ExplainerMethod.anchor_text:
+        if (
```
[the remark this answers is not in the document]
Comment (cliveseldon) [reply python/alibiexplainer/alibiexplainer/explainer.py:77]: Added by Black linter.

File: docs/samples/explanation/alibi/imagenet/README.md (hunk 1 above)
Comment (cliveseldon) [reply docs/samples/explanation/alibi/imagenet/README.md:29]: Have added link to docs.

**Comment by yuzisun** [comment]
Thanks @cliveseldon ! I think this should fix the e2e as well https://github.com/kubeflow/kfserving/pull/736.

/lgtm
/approve

**Pull request merged**